### PR TITLE
#182201727 (ft-book-accomodation-room): User should be able to book an accommodation room

### PR DESCRIPTION
**Pull request by uwambajeddy**
#### What does this PR do?
This PR allows the user to book a an accommodation room while creating a trip request

**Description of Task to be completed**
have the following endpoints working
`GET` `/api/v1/rooms` check available rooms
`POST` `/api/v1/user/trip` create a trip request
`GET` `/api/v1/user/trip/get` retrieve all of your trips
`Swagger document` `[server]/api/v1/docs`

**How should this be manually tested?**

- clone this repository
- checkout to this ft-book-accomodation-room-182201727
- run npm install to install dependencies
- use post man to test end points

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/182201727

### CODE REVIEW

**Review by Placide123 — DISMISSED** [review]

**Review by alaintresor — DISMISSED** [review]

File: src/database/migrations/20220611210302-create-trip-request.js [hunk 1]
```diff
@@ -61,6 +61,16 @@ module.exports = {
         onDelete: 'RESTRICT',
         onUpdate: 'CASCADE',
       },
+      roomId: {
```
Comment (joshwambere) [inline src/database/migrations/20220611210302-create-trip-request.js]: do not edit an already used migration

File: src/database/migrations/20220601094238-create-room.js [hunk 2]
```diff
@@ -20,6 +20,13 @@ export async function up(queryInterface, DataTypes) {
     accomodationId: {
       type: DataTypes.INTEGER,
     },
+    taken: {
```
Comment (joshwambere) [inline src/database/migrations/20220601094238-create-room.js]: do not edit an already used migrations

File: src/database/models/room.js [hunk 3]
```diff
@@ -30,6 +32,11 @@ export default (sequelize, DataTypes) => {
       bedType: DataTypes.STRING,
       bedCost: DataTypes.STRING,
```
Comment (joshwambere) [inline src/database/models/room.js]: rename to room cost or something that makes sense cause we book a whole room not only a bed

File: src/database/models/room.js [hunk 4]
```diff
@@ -30,6 +32,11 @@ export default (sequelize, DataTypes) => {
...
       bedDescription: DataTypes.STRING,
```
Comment (joshwambere) [inline src/database/models/room.js]: also should be a room description

File: src/database/models/room.js [hunk 5]
```diff
@@ -7,17 +7,19 @@ export default (sequelize, DataTypes) => {
      * This method is not a part of Sequelize lifecycle.
```
Comment (joshwambere) [inline src/database/models/room.js]: what is the use of UUID field in the room table?

**Review by joshwambere — CHANGES_REQUESTED** [review]

File: src/database/models/room.js (hunk 5 above)
Comment (uwambajeddy) [reply src/database/models/room.js]: I'm not the one who defined it, so I thought it have some other purposes that're out of my concern

File: src/database/models/room.js (hunk 4 above)
Comment (uwambajeddy) [reply src/database/models/room.js]: coy that

File: src/database/migrations/20220601094238-create-room.js (hunk 2 above)
Comment (uwambajeddy) [reply src/database/migrations/20220601094238-create-room.js]: I apologies

**Review by semy-tuyi — DISMISSED** [review]

**Review by mwisemarierose — DISMISSED** [review]

**Pull request merged**